projects/hslayers/components/layer-manager/widgets/scale-widget.component.html

```diff
@@ -1,36 +1,48 @@
-<div class="form-group" [hidden]="!isScaleVisible()"><!-- TODO: Remove function call from template -->
-  <label>{{ "COMMON.scale" | translate }}</label>
-  @if (minResolutionValid()) {
-    <span><span class="ms-1">{{ "COMMON.from" | translate }}</span> 1:{{
-    minResolution
-  }}</span>
-}
-@if (maxResolutionValid()) {
-  <span><span class="ms-1">{{ "COMMON.to" | translate }}</span> 1:{{
-  maxResolution
-}}</span>
-}
+<div class="hs-widget-scale">
+  <div class="hs-widget-header">
+    <div class="form-check">
+      <input type="checkbox" [id]="'scale-enabled-' + name" [checked]="scaleEnabled()" (change)="toggle()"
+        class="form-check-input" />
+      <label [for]="'scale-enabled-' + name" class="form-check-label">
+        {{ "LAYERMANAGER.layerEditor.scale" | translate }}
+      </label>
+    </div>
+  </div>
 
-<!-- TODO This editor need to be toggle-able-->
-<div class="form">
-  <br />
-  <label>{{ "COMMON.scale" | translate }}</label>
-  @if (minResolutionValid()) {
-    <div>
-      <div class="input-group">
-        <span class="input-group-text"><span>{{ "COMMON.from" | translate }}</span>&nbsp;1</span>
-        <input class="form-control" style="width: 65px" [(ngModel)]="minResolution" name="minResolution" />
+  @if (scaleEnabled()) {
+  <div class="hs-widget-content">
+    <div class="hs-scale-inputs">
+      <div class="hs-scale-input-group">
+        <label [for]="'min-scale-' + name" class="small text-muted">
+          {{ "COMMON.from" | translate }}
+        </label>
+        <div class="input-group input-group-sm">
+          <span class="input-group-text">1:</span>
+          <input type="number" [id]="'min-scale-' + name" class="form-control" [(ngModel)]="minResolution"
+            name="minResolution" [placeholder]="placeholder" min="0" step="1" />
+        </div>
       </div>
-    </div>
-  }
-  @if (maxResolutionValid()) {
-    <div>
-      <div class="input-group">
-        <span class="input-group-text">
-          <span>{{ "COMMON.to" | translate }}</span>&nbsp;1</span>
-          <input class="form-control" style="width: 65px" [(ngModel)]="maxResolution" name="maxResolution" />
+
+      <div class="hs-scale-input-group">
+        <label [for]="'max-scale-' + name" class="small text-muted">
+          {{ "COMMON.to" | translate }}
+        </label>
+        <div class="input-group input-group-sm">
+          <span class="input-group-text">1:</span>
+          <input type="number" [id]="'max-scale-' + name" class="form-control" [(ngModel)]="maxResolution"
+            name="maxResolution" [placeholder]="placeholder" min="1" step="1" />
         </div>
       </div>
+    </div>
+
+    @if (minResolutionValid() && maxResolutionValid()) {
```

Comment:
```suggestion
    @if (minResolutionValid() || maxResolutionValid()) {
```
Why not to show the summary in case any of the limits is valid?
That would be basically the same condition as `isScaleVisible()`. Perhaps we can use it directly...?
```suggestion
    @if (isScaleVisible()) {
```